Cannot run StarterPipeline generated by Dataflow wizard (Dataflow 2.0.0)

```
Exception in thread "main" java.lang.RuntimeException: Failed to construct instance from factory method DataflowRunner#fromOptions(interface org.apache.beam.sdk.options.PipelineOptions)
	at org.apache.beam.sdk.util.InstanceBuilder.buildFromMethod(InstanceBuilder.java:233)
	at org.apache.beam.sdk.util.InstanceBuilder.build(InstanceBuilder.java:162)
	at org.apache.beam.sdk.PipelineRunner.fromOptions(PipelineRunner.java:52)
	at org.apache.beam.sdk.Pipeline.create(Pipeline.java:141)
	at f.StarterPipeline.main(StarterPipeline.java:50)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.beam.sdk.util.InstanceBuilder.buildFromMethod(InstanceBuilder.java:222)
	... 4 more
Caused by: java.lang.IllegalArgumentException: Missing object or bucket in path: 'gs://my-awesome-bucket/', did you mean: 'gs://some-bucket/my-awesome-bucket'?
	at org.apache.beam.sdks.java.extensions.google.cloud.platform.core.repackaged.com.google.common.base.Preconditions.checkArgument(Preconditions.java:383)
	at org.apache.beam.sdk.extensions.gcp.storage.GcsPathValidator.verifyPath(GcsPathValidator.java:79)
	at org.apache.beam.sdk.extensions.gcp.storage.GcsPathValidator.validateOutputFilePrefixSupported(GcsPathValidator.java:62)
	at org.apache.beam.runners.dataflow.DataflowRunner.fromOptions(DataflowRunner.java:211)
	... 9 more
```

Arguments given to `main()` are as follows:
```
arg: --runner=DataflowRunner
arg: --project=my-gcp-project-id
arg: --gcpTempLocation=gs://my-awesome-bucket
arg: --stagingLocation=gs://my-awesome-bucket
```

This is the relevant bit:

Caused by: java.lang.IllegalArgumentException: Missing object or bucket in path: 'gs://my-awesome-bucket/', did you mean: 'gs://some-bucket/my-awesome-bucket'?

I seem to recall hitting this myself. May be a config issue.

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.core.test/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClientTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.core.test/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClientTest.java
@@ -35,6 +35,13 @@ public class GcsDataflowProjectClientTest {
   }
 
   @Test
+  public void testToGcsLocationUri_caseInsensitivePrefixDetection() {
+    assertEquals("gs://foo/bar", GcsDataflowProjectClient.toGcsLocationUri("GS://foo/bar"));
+    assertEquals("gs://foo/bar", GcsDataflowProjectClient.toGcsLocationUri("gS://foo/bar"));
+    assertEquals("gs://foo/bar", GcsDataflowProjectClient.toGcsLocationUri("Gs://foo/bar"));
+  }
+
+  @Test
   public void testToGcsLocationUriWithNullReturnsNull() {
     assertEquals(null, GcsDataflowProjectClient.toGcsLocationUri(null));
   }
@@ -54,6 +61,19 @@ public class GcsDataflowProjectClientTest {
   public void testToGcsLocationUriWithBucketNameOnlyReturnsWithPrefix() {
     String location = "foo-bar";
     assertEquals("gs://foo-bar", GcsDataflowProjectClient.toGcsLocationUri(location));
+  }
+
+  @Test
+  public void testToGcsBucketName_withoutObject() {
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("my-bucket"));
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("my-bucket/object"));
+  }
+
+  @Test
+  public void testToGcsBucketName_withObject() {
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("my-bucket/object"));
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("gs://my-bucket/object"));
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("GS://my-bucket/object"));
   }
 
   @Test

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.core.test/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClientTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.core.test/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClientTest.java
@@ -55,5 +55,18 @@ public class GcsDataflowProjectClientTest {
     String location = "foo-bar";
     assertEquals("gs://foo-bar", GcsDataflowProjectClient.toGcsLocationUri(location));
   }
+
+  @Test
+  public void testToGcsBucketName_stripsLeadingForwardSlashes() {
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("my-bucket"));
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("/my-bucket"));
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("///my-bucket/"));
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("///my-bucket/object"));
+
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("gs://my-bucket"));
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("gs:///my-bucket"));
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("gs://///my-bucket/"));
+    assertEquals("my-bucket", GcsDataflowProjectClient.toGcsBucketName("gs://///my-bucket/object"));
+  }
 }
 

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
@@ -90,15 +90,22 @@ public class GcsDataflowProjectClient {
     }
   }
 
-  public static String toGcsBucketName(String stagingLocation) {
-    String gcsLocation;
-    if (stagingLocation.toLowerCase(Locale.US).startsWith(GCS_PREFIX)) {
-      gcsLocation = stagingLocation.substring(GCS_PREFIX.length());
+  /**
+   * Extracts a bucket name from a given GCS URL. For example, {@code "/bucket/object"} returns
+   * {@code "bucket"}.
+   *
+   * @param gcsUrl GCS URL, which may or may not start with case-insensitive {@link #GCS_PREFIX}
+   * @return bucket name, which can be an empty string
+   */
+  public static String toGcsBucketName(String gcsUrl) {
+    String noPrefixUrl;
+    if (gcsUrl.toLowerCase(Locale.US).startsWith(GCS_PREFIX)) {
+      noPrefixUrl = gcsUrl.substring(GCS_PREFIX.length());
     } else {
-      gcsLocation = stagingLocation;
+      noPrefixUrl = gcsUrl;
     }
 
-    List<String> splitted = Splitter.on('/').omitEmptyStrings().splitToList(gcsLocation);
+    List<String> splitted = Splitter.on('/').omitEmptyStrings().splitToList(noPrefixUrl);
     return splitted.isEmpty() ? "" : splitted.get(0);
   }
 

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
@@ -92,7 +92,8 @@ public class GcsDataflowProjectClient {
 
   /**
    * Extracts a bucket name from a given GCS URL. For example, {@code "/bucket/object"} returns
-   * {@code "bucket"}.
+   * {@code "bucket"}. The method assumes that the input is a valid GCS URL. (It just returns
+   * the first segment split by {@code '/'}, ignoring leading {@code '/'}s and empty segments.)
    *
    * @param gcsUrl GCS URL, which may or may not start with case-insensitive {@link #GCS_PREFIX}
    * @return bucket name, which can be an empty string

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
@@ -21,6 +21,7 @@ import com.google.api.services.storage.Storage;
 import com.google.api.services.storage.model.Bucket;
 import com.google.api.services.storage.model.Buckets;
 import com.google.cloud.tools.eclipse.googleapis.IGoogleApiFactory;
+import com.google.common.base.CharMatcher;
 import com.google.common.base.Strings;
 import java.io.IOException;
 import java.util.List;
@@ -89,7 +90,14 @@ public class GcsDataflowProjectClient {
   }
 
   public static String toGcsBucketName(String stagingLocation) {
-    String gcsLocation = stripGcsPrefix(stagingLocation);
+    String gcsLocation;
+    if (stagingLocation.startsWith(GCS_PREFIX)) {
+      gcsLocation = stagingLocation.substring(GCS_PREFIX.length());
+    } else {
+      gcsLocation = stagingLocation;
+    }
+
+    gcsLocation = CharMatcher.is('/').trimLeadingFrom(gcsLocation);
 
     String bucketName;
     if (gcsLocation.indexOf('/') < 0) {
@@ -98,14 +106,6 @@ public class GcsDataflowProjectClient {
       bucketName = gcsLocation.substring(0, gcsLocation.indexOf('/'));
     }
     return bucketName;
-  }
-
-  public static String stripGcsPrefix(String gcsPath) {
-    if (gcsPath.startsWith(GCS_PREFIX)) {
-      return gcsPath.substring(GCS_PREFIX.length());
-    } else {
-      return gcsPath;
-    }
   }
 
   public static String toGcsLocationUri(String location) {

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
@@ -21,10 +21,11 @@ import com.google.api.services.storage.Storage;
 import com.google.api.services.storage.model.Bucket;
 import com.google.api.services.storage.model.Buckets;
 import com.google.cloud.tools.eclipse.googleapis.IGoogleApiFactory;
-import com.google.common.base.CharMatcher;
+import com.google.common.base.Splitter;
 import com.google.common.base.Strings;
 import java.io.IOException;
 import java.util.List;
+import java.util.Locale;
 import java.util.SortedSet;
 import java.util.TreeSet;
 import org.eclipse.core.runtime.IProgressMonitor;
@@ -91,26 +92,21 @@ public class GcsDataflowProjectClient {
 
   public static String toGcsBucketName(String stagingLocation) {
     String gcsLocation;
-    if (stagingLocation.startsWith(GCS_PREFIX)) {
+    if (stagingLocation.toLowerCase(Locale.US).startsWith(GCS_PREFIX)) {
       gcsLocation = stagingLocation.substring(GCS_PREFIX.length());
     } else {
       gcsLocation = stagingLocation;
     }
 
-    gcsLocation = CharMatcher.is('/').trimLeadingFrom(gcsLocation);
-
-    String bucketName;
-    if (gcsLocation.indexOf('/') < 0) {
-      bucketName = gcsLocation;
-    } else {
-      bucketName = gcsLocation.substring(0, gcsLocation.indexOf('/'));
-    }
-    return bucketName;
+    List<String> splitted = Splitter.on('/').omitEmptyStrings().splitToList(gcsLocation);
+    return splitted.isEmpty() ? "" : splitted.get(0);
   }
 
   public static String toGcsLocationUri(String location) {
-    if (Strings.isNullOrEmpty(location) || location.startsWith(GCS_PREFIX)) {
+    if (Strings.isNullOrEmpty(location)) {
       return location;
+    } else if (location.toLowerCase(Locale.US).startsWith(GCS_PREFIX)) {
+      return GCS_PREFIX + location.substring(GCS_PREFIX.length());
     }
     return GCS_PREFIX + location;
   }

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java
@@ -88,13 +88,9 @@ public class GcsDataflowProjectClient {
     }
   }
 
-  private static String toGcsBucketName(String stagingLocation) {
-    String gcsLocation;
-    if (stagingLocation.startsWith(GCS_PREFIX)) {
-      gcsLocation = stagingLocation.substring(GCS_PREFIX.length());
-    } else {
-      gcsLocation = stagingLocation;
-    }
+  public static String toGcsBucketName(String stagingLocation) {
+    String gcsLocation = stripGcsPrefix(stagingLocation);
+
     String bucketName;
     if (gcsLocation.indexOf('/') < 0) {
       bucketName = gcsLocation;
@@ -102,6 +98,14 @@ public class GcsDataflowProjectClient {
       bucketName = gcsLocation.substring(0, gcsLocation.indexOf('/'));
     }
     return bucketName;
+  }
+
+  public static String stripGcsPrefix(String gcsPath) {
+    if (gcsPath.startsWith(GCS_PREFIX)) {
+      return gcsPath.substring(GCS_PREFIX.length());
+    } else {
+      return gcsPath;
+    }
   }
 
   public static String toGcsLocationUri(String location) {

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponentTest.java
@@ -16,6 +16,7 @@
 
 package com.google.cloud.tools.eclipse.dataflow.ui.preferences;
 
+import static org.junit.Assert.assertTrue;
 import static org.mockito.Matchers.anyString;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
@@ -171,5 +172,14 @@ public class RunOptionsDefaultsComponentTest {
       Thread.sleep(50);
     }
     Assert.assertArrayEquals(buckets, combo.getItems());
+  }
+
+  @Test
+  public void testBucketNameStatus_gscPathWithObjectIsOk() {
+    component.setStagingLocationText("bucket/object");
+    assertTrue(component.bucketNameStatus().isOK());
+
+    component.setStagingLocationText("gs://bucket/object");
+    assertTrue(component.bucketNameStatus().isOK());
   }
 }

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java
@@ -44,7 +44,6 @@ import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.NullProgressMonitor;
-import org.eclipse.core.runtime.Status;
 import org.eclipse.jface.wizard.WizardPage;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.events.FocusAdapter;
@@ -260,18 +259,17 @@ public class RunOptionsDefaultsComponent {
       verifyJob.cancel();
     }
 
-    IStatus status = validateStagingLocation();
-    if (!status.isOK()) {
-      messageTarget.setError(status.getMessage());
-      return;
-    }
-
     final String bucketNamePart = extractBucketNamePart();
-    // Validation may succeed with an empty bucket name (but only when there is no GCS object name).
     if (bucketNamePart.isEmpty()) {
       // If the bucket name is empty, we don't have anything to verify; and we don't have any
       // interesting messaging.
       setPageComplete(true);
+      return;
+    }
+
+    IStatus status = bucketNameStatus();
+    if (!status.isOK()) {
+      messageTarget.setError(status.getMessage());
       return;
     }
 
@@ -297,16 +295,17 @@ public class RunOptionsDefaultsComponent {
               }
 
               if (result.accessible) {
-                messageTarget.setInfo("Verified bucket %s is accessible." + bucketNamePart);
+                messageTarget.setInfo(
+                    String.format("Verified bucket %s is accessible.", bucketNamePart));
                 createButton.setEnabled(false);
                 setPageComplete(true);
               } else {
-                messageTarget.setError(String.format("Couldn't fetch bucket %s", bucketNamePart));
+                messageTarget.setError(String.format("Couldn't fetch bucket %s.", bucketNamePart));
                 createButton.setEnabled(true);
                 setPageComplete(false);
               }
             } catch (InterruptedException | ExecutionException e) {
-              messageTarget.setError(String.format("Couldn't fetch bucket %s", bucketNamePart));
+              messageTarget.setError(String.format("Couldn't fetch bucket %s.", bucketNamePart));
               setPageComplete(false);
             }
           }
@@ -396,23 +395,16 @@ public class RunOptionsDefaultsComponent {
     return GcsDataflowProjectClient.toGcsBucketName(stagingLocationInput.getText());
   }
 
-  private IStatus validateStagingLocation() {
-    String bucketName = extractBucketNamePart();
-    if (bucketName.isEmpty()) {
-      String gcsPath = GcsDataflowProjectClient.stripGcsPrefix(stagingLocationInput.getText());
-      if (!gcsPath.isEmpty()) {
-        // Object part is not null; for example, input value is "/something" or 'gs:///something'.
-        return new Status(IStatus.ERROR, DataflowUiPlugin.PLUGIN_ID, "Bucket name part is empty.");
-      }
-    }
-    return bucketNameValidator.validate(bucketName);
+  @VisibleForTesting
+  IStatus bucketNameStatus() {
+    return bucketNameValidator.validate(extractBucketNamePart());
   }
 
   private class EnableCreateButton implements ModifyListener {
 
     @Override
     public void modifyText(ModifyEvent event) {
-      boolean enabled = !extractBucketNamePart().isEmpty() && validateStagingLocation().isOK();
+      boolean enabled = !extractBucketNamePart().isEmpty() && bucketNameStatus().isOK();
       createButton.setEnabled(enabled);
     }
 

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java
@@ -392,6 +392,7 @@ public class RunOptionsDefaultsComponent {
   private static final BucketNameValidator bucketNameValidator = new BucketNameValidator();
 
   private String extractBucketNamePart() {
+    // Don't trim text unless you consistently trim everywhere else (e.g., "getStagingLocation()").
     return GcsDataflowProjectClient.toGcsBucketName(stagingLocationInput.getText());
   }
 

--- a/plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java
+++ b/plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java
@@ -39,12 +39,12 @@ import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableSortedSet;
-import java.util.Locale;
 import java.util.SortedSet;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Status;
 import org.eclipse.jface.wizard.WizardPage;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.events.FocusAdapter;
@@ -123,7 +123,7 @@ public class RunOptionsDefaultsComponent {
 
     Label comboLabel = new Label(target, SWT.NULL);
     stagingLocationInput = new Combo(target, SWT.DROP_DOWN);
-    createButton = ButtonFactory.newPushButton(target, "&Create");
+    createButton = ButtonFactory.newPushButton(target, "&Create Bucket");
     createButton.setEnabled(false);
 
     accountSelector.selectAccount(preferences.getDefaultAccountEmail());
@@ -259,16 +259,19 @@ public class RunOptionsDefaultsComponent {
       // FIXME: this has no effect, as "VerifyStagingLocationJob" doesn't honor cancellation.
       verifyJob.cancel();
     }
-    if (trimBucketName().isEmpty()) {
-      // If the staging location is empty, we don't have anything to verify; and we don't have any
-      // interesting messaging.
-      setPageComplete(true);
+
+    IStatus status = validateStagingLocation();
+    if (!status.isOK()) {
+      messageTarget.setError(status.getMessage());
       return;
     }
 
-    IStatus status = bucketNameStatus();
-    if (!status.isOK()) {
-      messageTarget.setError(status.getMessage());
+    final String bucketNamePart = extractBucketNamePart();
+    // Validation may succeed with an empty bucket name (but only when there is no GCS object name).
+    if (bucketNamePart.isEmpty()) {
+      // If the bucket name is empty, we don't have anything to verify; and we don't have any
+      // interesting messaging.
+      setPageComplete(true);
       return;
     }
 
@@ -294,16 +297,16 @@ public class RunOptionsDefaultsComponent {
               }
 
               if (result.accessible) {
-                messageTarget.setInfo("Found staging location " + stagingLocation);
+                messageTarget.setInfo("Verified bucket %s is accessible." + bucketNamePart);
                 createButton.setEnabled(false);
                 setPageComplete(true);
               } else {
-                messageTarget.setError(String.format("Couldn't fetch bucket %s", stagingLocation));
+                messageTarget.setError(String.format("Couldn't fetch bucket %s", bucketNamePart));
                 createButton.setEnabled(true);
                 setPageComplete(false);
               }
             } catch (InterruptedException | ExecutionException e) {
-              messageTarget.setError(String.format("Couldn't fetch bucket %s", stagingLocation));
+              messageTarget.setError(String.format("Couldn't fetch bucket %s", bucketNamePart));
               setPageComplete(false);
             }
           }
@@ -389,23 +392,27 @@ public class RunOptionsDefaultsComponent {
 
   private static final BucketNameValidator bucketNameValidator = new BucketNameValidator();
 
-  private String trimBucketName() {
-    String bucketName = stagingLocationInput.getText().trim();
-    if (bucketName.toLowerCase(Locale.US).startsWith("gs://")) {
-      bucketName = bucketName.substring(5);
-    }
-    return bucketName;
+  private String extractBucketNamePart() {
+    return GcsDataflowProjectClient.toGcsBucketName(stagingLocationInput.getText());
   }
 
-  private IStatus bucketNameStatus() {
-    return bucketNameValidator.validate(trimBucketName());
+  private IStatus validateStagingLocation() {
+    String bucketName = extractBucketNamePart();
+    if (bucketName.isEmpty()) {
+      String gcsPath = GcsDataflowProjectClient.stripGcsPrefix(stagingLocationInput.getText());
+      if (!gcsPath.isEmpty()) {
+        // Object part is not null; for example, input value is "/something" or 'gs:///something'.
+        return new Status(IStatus.ERROR, DataflowUiPlugin.PLUGIN_ID, "Bucket name part is empty.");
+      }
+    }
+    return bucketNameValidator.validate(bucketName);
   }
 
   private class EnableCreateButton implements ModifyListener {
 
     @Override
     public void modifyText(ModifyEvent event) {
-      boolean enabled = !trimBucketName().isEmpty() && bucketNameStatus().isOK();
+      boolean enabled = !extractBucketNamePart().isEmpty() && validateStagingLocation().isOK();
       createButton.setEnabled(enabled);
     }
 

